tools/generate-wire.py

```diff
@@ -338,20 +354,30 @@ def print_fromwire(self, is_header):
 
                 self.print_fromwire_array(subcalls, basetype, f, '*' + f.name,
                                           f.lenvar)
-            elif f.is_assignable():
-                subcalls.append("\t//3th case {name}".format(name=f.name))
-                if f.is_len_var:
-                    subcalls.append('\t{} = fromwire_{}(&cursor, &plen);'
+            else:
+                if f.is_optional():
+                    subcalls.append("\tif (!fromwire_bool(&cursor, &plen))\n"
+                                    "\t\t*{} = NULL;\n"
+                                    "\telse {{\n"
+                                    "\t\t*{} = tal(ctx, {});\n"
+                                    "\t\tfromwire_{}(&cursor, &plen, *{});\n"
+                                    "\t}}"
+                                    .format(f.name, f.name, f.fieldtype.name,
```

Comment:
nit: I wonder if it would be a good idea to have named placeholders here, considering the number of arguments. It might be good in a few other places in this patch as well.

Reply:
Yes, this tool could definitely use some love!